frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,196 @@
+          this.isAdmin = user['isAdministrator'];
+          this.hasOrganization = user['organizations'].length > 0;
+        }
+     });
+
+    this.activatedRoute.params
+    .map(params => params.offerId)
+    .filter(offerId => offerId !== undefined)
+    .switchMap(offerId => this.offersService.getOffer(offerId))
+    .do(offer => this.toDataUrl(offer.image, (base64) => this.offer.image = {
+      content: base64,
+      filename: 'image.jpg'
+    }))
+    .subscribe((response: BaseOffer) => {
+      this.offer = response as AppOffer;
+      this.inEditMode = true;
+      this.offer = this.removeTime();
+      this.form.patchValue(this.offer);
+    });
+  }
+
+// Temporary fix to make editing work properly, to delete after we fix datetime issue
+  removeTime() {
+    this.offer.startedAt = this.offer.startedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    this.offer.finishedAt = this.offer.finishedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    if (this.offer.recruitmentEndDate) {
+      this.offer.recruitmentEndDate = this.offer.recruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if (this.offer.recruitmentStartDate) {
+      this.offer.recruitmentStartDate = this.offer.recruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if (this.offer.reserveRecruitmentEndDate) {
+      this.offer.reserveRecruitmentEndDate = this.offer.reserveRecruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    if (this.offer.reserveRecruitmentStartDate) {
+      this.offer.reserveRecruitmentStartDate = this.offer.reserveRecruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    return this.offer;
+  }
+
+  // TODO - delete when we decide what date format we want to have
+  addTime() {
+    this.form.value.startedAt = this.form.value.startedAt + 'T00:00:00Z';
+    this.form.value.finishedAt = this.form.value.finishedAt + 'T00:00:00Z';
+
+    if (this.form.value.recruitmentEndDate) {
+      this.form.value.recruitmentEndDate = this.form.value.recruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.recruitmentStartDate) {
+      this.form.value.recruitmentStartDate = this.form.value.recruitmentStartDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentEndDate) {
+      this.form.value.reserveRecruitmentEndDate = this.form.value.reserveRecruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentStartDate) {
+      this.form.value.reserveRecruitmentStartDate = this.form.value.reserveRecruitmentStartDate + 'T00:00:00Z';
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.userSubscription) {
+      this.userSubscription.unsubscribe();
+    };
+  }
+
+  toDataUrl(url, callback) {
+    const reader = new FileReader();
+    this.http.get(url, {responseType: 'blob'})
+    .subscribe(response => {
+      reader.onloadend = function() {
+          callback(reader.result)
```

Comment:
Instead of using callback here, write the logic to update the image conent as you did in onFileSelected() method